### PR TITLE
correct name in comment

### PR DESCRIPTION
**Pull request by darius**

### CODE REVIEW

**Comment by darius** [comment]
The pull-request title is misleading -- this collects several unrelated patches that I think all make the code clearer. (Sorry, I'd meant them to be separate but forgot that github would funnel my later commits into the same pull request.)

I've tested by rerunning the provided tests, but not otherwise.

File: gitlet.js [hunk 1]
```diff
@@ -1903,21 +1887,22 @@ var status = {
...
+
+    // Gather all the sections, keeping only nonempty ones, and flatten them
+    // together into a string.
+    return util.flatten(["On branch " + refs.headBranchName(),
+                         listing("Untracked files:", untracked()),
+                         listing("Unmerged paths:", index.conflictedPaths()),
+                         listing("Changes to be committed:", toBeCommitted()),
+                         listing("Changes not staged for commit:", notStagedForCommit())])
+        .join("\n");
   }
```
Comment (maryrosecook) [inline gitlet.js]: I love this!

File: gitlet.js [hunk 2]
```diff
@@ -1952,16 +1937,15 @@ var parseOptions = function(argv) {
...
-      .apply(null, new Array(fn.length - commandArgs.length - 1))
-      .map(function() { return undefined; });
-
-  return fn.apply(gitlet, commandArgs.concat(unspecifiedArgs).concat(rawArgs));
+  var commandArgs = opts._.slice(3);
+  while (commandArgs.length < fn.length - 1) {
+    commandArgs.push(undefined);
+  }
+  return fn.apply(gitlet, commandArgs.concat(opts));
 };
```
Comment (maryrosecook) [inline gitlet.js]: And I love this!

**Comment by maryrosecook** [comment]
Thanks so much for this, Darius! It's great.  I've integrated your changes.

**Comment by darius** [comment]
You're welcome! And thanks for this project -- it's super-clear and educational.

**Comment by maryrosecook** [comment]
My pleasure!

--
http://maryrosecook.com

**Pull request merged**